### PR TITLE
Improve Transaction Manager logging

### PR DESCRIPTION
**Pull request by adamretter**

### CODE REVIEW

**Comment by dizzzz** [comment]
You could consider using String.format to make nicely readable texts?

**Comment by adamretter** [comment]
@dizzzz Somewhere particular? I really just changed the logging priorities and added checks around the debug log levels for performance

File: src/org/exist/storage/txn/TransactionManager.java [hunk 1]
```diff
@@ -161,12 +161,14 @@ public Txn beginTransaction() {
 
 			public Txn execute() {
 				final long txnId = nextTxnId++;
-                LOG.debug("Starting new transaction: " + txnId);
+                if(LOG.isDebugEnabled()) {
+                    LOG.debug("Starting new transaction: " + txnId);
```
Comment (dizzzz) [inline src/org/exist/storage/txn/TransactionManager.java]: Here

File: src/org/exist/storage/txn/TransactionManager.java [hunk 2]
```diff
@@ -203,7 +205,9 @@ public Object execute() {
                 txn.releaseAll();
                 transactions.remove(txn.getId());
                 processSystemTasks();
-                LOG.debug("Committed transaction: " + txn.getId());
+                if(LOG.isDebugEnabled()) {
+                    LOG.debug("Committed transaction: " + txn.getId());
```
Comment (dizzzz) [inline src/org/exist/storage/txn/TransactionManager.java]: Here

File: src/org/exist/storage/txn/TransactionManager.java [hunk 3]
```diff
@@ -222,7 +226,7 @@ public Object execute() {
                 try {
                     journal.writeToLog(new TxnAbort(txn.getId()));
                 } catch (final TransactionException e) {
-                    LOG.warn("Failed to write abort record to journal: " + e.getMessage());
+                    LOG.error("Failed to write abort record to journal: " + e.getMessage());
```
Comment (dizzzz) [inline src/org/exist/storage/txn/TransactionManager.java]: Here

File: src/org/exist/storage/txn/TransactionManager.java [hunk 4]
```diff
@@ -309,14 +311,16 @@ public void reindex(final DBBroker broker) {
         try {
             broker.reindexCollection(XmldbURI.ROOT_COLLECTION_URI);
         } catch (final PermissionDeniedException e) {
-            LOG.warn("Exception during reindex: " + e.getMessage(), e);
+            LOG.error("Exception during reindex: " + e.getMessage(), e);
```
Comment (dizzzz) [inline src/org/exist/storage/txn/TransactionManager.java]: Here

File: src/org/exist/storage/txn/TransactionManager.java [hunk 5]
```diff
@@ -309,14 +311,16 @@ public void reindex(final DBBroker broker) {
...
         } finally {
         	broker.popSubject();
         }
     }
 
     public void shutdown() {
-        LOG.debug("Shutting down transaction manager. Uncommitted transactions: " + transactions.size());
+        if(LOG.isDebugEnabled()) {
+            LOG.debug("Shutting down transaction manager. Uncommitted transactions: " + transactions.size());
```
Comment (dizzzz) [inline src/org/exist/storage/txn/TransactionManager.java]: Her.e

File: src/org/exist/storage/txn/TransactionManager.java [hunk 6]
```diff
@@ -335,8 +339,7 @@ private int uncommittedTransaction() {
             {return count;}
         for (final Map.Entry<Long, TxnCounter> entry : transactions.entrySet()) {
             if (entry.getValue().counter > 0) {
-                LOG.warn("Found an uncommitted transaction with id " + entry.getKey() + ". Pending operations: " +
-                    entry.getValue().counter);
+                LOG.warn("Found an uncommitted transaction with id " + entry.getKey() + ". Pending operations: " + entry.getValue().counter);
```
Comment (dizzzz) [inline src/org/exist/storage/txn/TransactionManager.java]: For sure here

**Comment by dizzzz** [comment]
Well it is not that important.... I'll pull in

**Comment by adamretter** [comment]
@dizzzz String.format is considerably slower than `+`, so typically I only use it where I actually need to apply formatting to the values being appended to the String. In each of the cases you have highlighted we don't need to do any explicit formatting, instead we just need concatenation. For ref see - http://stackoverflow.com/questions/925423/is-it-better-practice-to-use-string-format-over-string-concatenation-in-java

**Comment by dizzzz** [comment]
Yes I know, but for not criticAl performance code I prefer  readability :)

**Comment by adamretter** [comment]
@dizzzz fair enough

**Pull request merged**